opslevel/resource_opslevel_team.go

```diff
@@ -195,7 +195,11 @@ func (teamResource *TeamResource) Read(ctx context.Context, req resource.ReadReq
 	}
 
 	team, err := teamResource.client.GetTeam(opslevel.ID(stateModel.Id.ValueString()))
-	if err != nil || team == nil {
+	if team == nil && opslevel.HasBadHttpStatus(err) {
+		resp.Diagnostics.AddError("HTTP status error", fmt.Sprintf("unable to read team, got error: %s", err))
+		return
+	}
+	if err != nil {
 		resp.Diagnostics.AddWarning("State drift", stateResourceMissingMessage("opslevel_team"))
 		resp.State.RemoveResource(ctx)
 		return
```

Comment:
This would only cause us to recreate the team though if no error at all happens which isn't what we want.  I think we need something like:

```
	if (team == nil || team.Id == nil) && !opslevel.HasBadHttpStatus(err) {
		resp.State.RemoveResource(ctx)
		return
	}
	if err != nil {
		resp.Diagnostics.AddError("Error", fmt.Sprintf("unable to read team, got error: %s", err))
		return
	}
```